apps/browser-extension-wallet/src/views/browser-view/features/sing-message/SignMessage.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useState } from 'react';
+import styles from './SignMessage.module.scss';
+import { useTranslation } from 'react-i18next';
+import { WalletOwnAddressDropdown, AddressSchema } from '@lace/core';
+import { Drawer, DrawerNavigation, TextArea, Button, toast, PostHogAction } from '@lace/common';
+import { Text } from '@input-output-hk/lace-ui-toolkit';
+import { Cip30DataSignature } from '@cardano-sdk/dapp-connector';
+import { SignMessagePassword } from '@views/browser/features/sing-message/SignMessagePassword';
+import { Password } from '@input-output-hk/lace-ui-toolkit/dist/design-system/password-box/uncontrolled-password-box-input.component';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import { useAnalyticsContext } from '@providers';
+
+export type SignMessageProps = {
+  addresses: AddressSchema[];
+  visible?: boolean;
+  onClose: () => void;
+  onSign: (address: string, message: string, password: string) => void;
+  isSigningInProgress: boolean;
+  signedMessage: Cip30DataSignature | null;
+  error: string | null;
+  hardwareWalletError: string | null;
+  isHardwareWallet: boolean;
+};
+
+export const SignMessage = ({
+  addresses = [],
+  visible = true,
+  onClose,
+  onSign,
+  isSigningInProgress,
+  signedMessage,
+  error,
+  hardwareWalletError,
+  isHardwareWallet
+}: SignMessageProps): React.ReactElement => {
+  const { t } = useTranslation();
+  const analytics = useAnalyticsContext();
+
+  const [selectedAddress, setSelectedAddress] = useState<string>('');
+  const [shouldShowPasswordPrompt, setShouldShowPasswordPrompt] = useState<boolean>(false);
+  const [password, setPassword] = useState<string>('');
```

Comment:
Using this hook will help clearing secrets from memory. We spent a lot of time figuring out how to make it work and get around React cache. The hook also returns a `clearSecrets` function that should be called when we are done.

```suggestion
  const { password, setPassword } = useSecrets();
```